### PR TITLE
Use `dfn` elements for defining terms

### LINKED ISSUE #1664: Wrap defined term in `dfn` elements
Our definitions usually contain the defined term wrapped in underscores (hence in italic), e.g.
```markdown
A _semantic link_ is any element which has the [semantic role][] of [`link`][link] or a [semantic role][] that inherits from the [`link`][link] role.
```
We should rather wrap them if `dfn` elements (if our markdown processing allows it), e.g.
```markdown
A <dfn>semantic link<dfn> is any element which has the [semantic role][] of [`link`][link] or a [semantic role][] that inherits from the [`link`][link] role.
```
We should also add an `id` to the `dfn` element either all the time or at least for those definitions that define more than one term (e.g. [Text Inheriting its Programmatic Language from an Element](https://github.com/act-rules/act-rules.github.io/blob/develop/pages/glossary/text-inheriting-language.md) so that we can refer the correct one when needed.

### PR DESCRIPTION
**Pull request by Jym77**
Showcasing how we could use `dfn` elements for defining terms in definitions and conditional lists.
Some of the changes are not really needed but used to illustrate some possibilities 😄 

In order to ensure name uniqueness (the definitions are included inside the rules), I namespace the IDs in definition with the definition key, as far as I understand the `:` in an ID shouldn't be a problem. (otherwise we can just namespace with `_`). IDs in rules are namespaced by the rule ID.

Pros:
* Semantically correct 😁 
* Can link to the precise part of the definition that is of interest.
* Can backlink to a precise condition in a conditional list.

Closes issue(s):

- closes #1664 

Need for Call for Review:
This will require a 1 week Call for Review (should be purely cosmetic, but could be touching a lot of stuff).

---

### **When creating PR:**

- [X] Make sure you're requesting to **pull a branch** (right side) to the `develop` branch (left side).
- [X] Make sure you do not remove the "How to Review and Approve" section in your pull request description

### **After creating PR:**

- [X] Add yourself (and co-authors) as "Assignees" for PR.
- [X] Add label to indicate if it's a `Rule`, `Definition` or `Chore`.
- [X] [Link the PR to any issue it solves](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue). This will be done automatically by [referencing the issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) at the top of this comment in the indicated place.
- [x] Optionally request feedback from anyone in particular by assigning them as "Reviewers".

### **When merging a PR:**

- [ ] Close any issue that the PR resolves. This will happen automatically upon merging if the PR was correctly linked to the issue, e.g. by referencing the issue at the top of this comment.

## How to Review And Approve

- Go to the “Files changed” tab
- Here you will have the option to leave comments on different lines.
- Once the review is completed, find the “Review changes” button in the top right, select “Approve” (if you are really confident in the rule) or "Request changes" and click “Submit review”.
- Make sure to also review the proposed Call for Review period. In case of disagreement, the longer period wins.

### CODE REVIEW

File: pages/glossary/text-inheriting-language.md [hunk 1]
```diff
@@ -9,13 +9,13 @@ input_aspects:
   - CSS Styling
 ---
 
-The _text inheriting its programmatic language_ from an element E is composed of all the following texts:
+The <dfn id="text-inheriting-language:text">text inheriting its programmatic language</dfn> from an element E is composed of all the following texts:
 
 - **text nodes**: the value of any [text nodes][] that are [visible][] or [included in the accessibility tree][] and children of an element inheriting its programmatic language from E;
-- **accessible text**: the [accessible name][] and [accessible description][] of any element inheriting its programmatic language from E, and [included in the accessibility tree][];
+- **accessible text**: the [accessible name][] and [accessible description][] of any [element inheriting its programmatic language](#text-inheriting-language:element) from E, and [included in the accessibility tree][];
```
Comment (Jym77) [inline pages/glossary/text-inheriting-language.md:15]: Definition with subterms can have internal link to the correct part.

**Comment by Jym77** [comment]
Putting this back on Agenda so we can discuss (follow up to https://www.w3.org/2021/10/14-act-r-minutes.html#t05) and decide whether it is something we want to do or not…

**Review by HelenBurge — APPROVED** [review]

**Review by daniel-montalvo — APPROVED** [review]

**Review by kherr-a11y — APPROVED** [review]

File: _rules/element-lang-matches-default-language-off6ek.md [hunk 2]
```diff
@@ -35,10 +35,10 @@ htmlHintIgnore:
 
 This rule applies to any [HTML element][] with a `lang` attribute for which all the following are true:
 
-- the element is an [inclusive descendant][] in the [flat tree][] of a `body` element; and
-- the element is in a [document][] with a [content type][] of `text/html`; and
-- the element's `lang` [attribute value][] is a [valid language tag][]; and
-- there is some non-empty [text inheriting its programmatic language][] from the element.
+- <dfn id="in-body">**(in body)**</dfn> the element is an [inclusive descendant][] in the [flat tree][] of a `body` element; and
```
Comment (WilcoFiers) [inline _rules/element-lang-matches-default-language-off6ek.md:38]:
```suggestion
- <dfn id="in-body">in body</dfn>: the element is an [inclusive descendant][] in the [flat tree][] of a `body` element; and
```

**Comment by Jym77** [comment]
Add doc on writing defs and listed conditions.

**Comment by Jym77** [comment]
Namespace ids in rules by the rule ID.

**Comment by kengdoj** [comment]
No objection to this change inside the rules. 

This is a while ago, but I recall discussion of a single ACT glossary and taking definitions out of rules. Is that no longer being considered?

**Review by WilcoFiers — APPROVED** [review]

**Review by carlosapaduarte — APPROVED** [review]

**Comment by Jym77** [comment]
Call for Reviews ends on February 10.

**Review by JenniferChadwick — APPROVED** [review]

**Review by colabottles — APPROVED** [review]

**Comment by Jym77** [comment]
Call for Review has ended, merging.

**Pull request merged**